apps/aecontract/src/aect_call_tx.erl

```diff
@@ -104,6 +123,46 @@ new(#{caller_id   := CallerId,
                           },
     {ok, aetx:new(?MODULE, Tx)}.
 
+-spec from_db_format(tx() | tuple()) -> tx().
+from_db_format(#contract_call_tx{} = Tx) -> Tx;
+from_db_format(Tuple) ->
+    case setelement(1, Tuple, db_contract_call_tx) of
+        #db_contract_call_tx{
+           caller_id        = CallerId,
+           nonce            = Nonce,
+           contract_id      = ContractId,
+           abi_version      = ABI,
+           fee              = Fee,
+           ttl              = TTL,
+           amount           = Amount,
+           gas              = Gas,
+           gas_price        = GasPrice,
+           call_data        = CallData,
+           call_stack       = CallStack
+          } ->
+            CallOrigin =
+                case aec_id:specialize(CallerId) of
+                    {contract, Pubkey} -> Pubkey;
+                    {account,  Pubkey} -> Pubkey
+                end,
+            #contract_call_tx{
+               caller_id        = CallerId,
+               nonce            = Nonce,
+               contract_id      = ContractId,
+               abi_version      = ABI,
+               fee              = Fee,
+               ttl              = TTL,
+               amount           = Amount,
+               gas              = Gas,
+               gas_price        = GasPrice,
+               call_data        = CallData,
+               call_stack       = CallStack,
+               call_origin      = CallOrigin
+              };
+        _ ->
```

Comment:
is there a risk that one would call this function once more after the first conversion and then fail with an error?

Reply:
It will then be caught by the previous function clause.